torch/_inductor/fx_passes/post_grad.py

```diff
@@ -1006,6 +1007,46 @@ def repl(inp, mat1, mat2):
     match.replace_by_example(repl, [inp, mat1, mat2])
 
 
+def register_partial_reduction_pattern():
+    "Reuse partial reductions in complete reductions"
+
+    # post grad equivalents
+    equiv_red = {
+        aten.amax.default: aten.max.default,
+        aten.amin.default: aten.min.default,
+    }
+
+    # TODO - could support more reductions. they have different signatures which makes
+    # this a bit more annoying
+    for red_op in (aten.amax.default, aten.amin.default):
```

Comment:
would this also work for `max` and `min`?

Reply:
`max` and `min` don't support reducing over multiple dims, and when you specify a particular dim, it returns both the index and the value so would need a bit of extra work.